zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepoSync.java

```diff
@@ -246,14 +268,17 @@ NotebookRepo getRepo(int repoIndex) throws IOException {
         /* note exists in source and destination storage systems */
         sdate = lastModificationDate(sourceRepo.get(snote.getId(), null));
         ddate = lastModificationDate(destRepo.get(dnote.getId(), null));
-        if (sdate.after(ddate)) {
-          /* source contains more up to date note - push */
-          pushIDs.add(snote.getId());
-          LOG.info("Modified note is added to push list : " + sdate);
-        } else if (sdate.compareTo(ddate) != 0) {
-          /* destination contains more up to date note - pull */
-          LOG.info("Modified note is added to pull list : " + ddate);
-          pullIDs.add(snote.getId());
+
+        if (sdate.compareTo(ddate) != 0) {
+          if (sdate.after(ddate) || oneWaySync) {
+            /* source contains more up to date note - push */
```

Comment:
I think it's better to add comment here that basically everything will be pushed in case of `oneWaySync`